cocotb/triggers.py

```diff
@@ -52,7 +52,36 @@
 class TriggerException(Exception):
     pass
 
-class Trigger(_py_compat.with_metaclass(abc.ABCMeta)):
+
+class TriggerHelper(object):
+    """A base class with commen methods for event type classes"""
+
+    @decorators.coroutine
+    def aslongas(self, condition, readonly=True):
```

Comment:
``while`` is typically used as the counterpart to ``until`` - any specific reason as to why are not using that?

Reply:
+1, but needs to be `while_` to avoid conflicts. Underscore suffix is standard to avoid keyword clashes

Reply:
Unfortunately, the rest of cocotb uses a leading underscore...
Regardless of prefix or suffix, the "until" should maybe get the same treatment for consistency.

Reply:
To me prefix underscore means things not to be used by user code so I will put it as suffix.

Reply:
Not sure what you're referring to @cmarqu: `_name` is for private fields, `name_` is for public fields which clash with keywords.

Reply:
I'm not sure we need `aslongas`/ `while` at all - all the use cases I can see about use `not` anyway, so would be clearer spelt with `until`.

Reply:
The difference is that while does not wait for the event if condition is True and until does first yield the event and then check the condition; e.g. latter is modeled after `do ... until`.

Reply:
> Not sure what you're referring to @cmarqu: `_name` is for private fields, `name_` is for public fields which clash with keywords.

I mean these: https://github.com/cocotb/cocotb/blob/a62759901c30db72b592da6c559dbe71efa5b518/cocotb/handle.py#L67-L70

Reply:
I used `not` to keep with old code to interpret the `.value` as `boolean`; could also have used `== 0`. Actually did the former on first internal iteration.

Reply:
@cmarqu, that's a subtly different case, and _does_ match the built-in `namedtuple` behavior - so I don't think matters.

Reply:
> I used `not` to keep with old code to interpret the `.value` as `boolean`; could also have used `== 0`. Actually did the former on first internal iteration.

Don't think there any need to make the lambda return a boolean. It's used eventually in a boolean context anyway.

Reply:
New iteration of patch will not use ``while`` anymore but methods ``until``, ``until_after``, ``do_until`` and ``do_until_after``